src/miral/launch_app.cpp

```diff
@@ -29,29 +29,69 @@
 
 namespace
 {
-auto mir_env_variables() -> std::vector<std::string>
+class Environment
 {
-    static char const mir_prefix[] = "MIR_";
+public:
+    Environment();
+
+    void setenv(std::string const& name, std::string const& value);
+    void unsetenv(std::string const& name);
+    auto exec_env() const & -> std::vector<char const*>;
+private:
+    std::vector<std::string> env_strings;
+};
 
-    std::vector<std::string> vars_to_remove;
+Environment::Environment()
+{
+    static char const mir_prefix[] = "MIR_";
 
     for (auto var = environ; *var; ++var)
     {
         auto const var_begin = *var;
-        if (strncmp(var_begin, mir_prefix, sizeof(mir_prefix) - 1) == 0)
+        if (strncmp(var_begin, mir_prefix, sizeof(mir_prefix) - 1) != 0)
+        {
+            env_strings.emplace_back(var_begin);
+        }
+    }
+}
+
+void Environment::setenv(std::string const& name, std::string const& value)
+{
+    auto const entry = name + "=" + value;
+
+    for (auto it = env_strings.begin(); it != env_strings.end(); ++it)
```

Comment:
This could be
```
for (auto& it : env_strings)
```
right? (plus the associated pointer->reference changes)

I'm not missing some C++ subtlety?

Reply:
Right